codegen/src/main/java/org/nd4j/codegen/cli/CLI.java

```diff
@@ -53,32 +118,15 @@ public void runMain(String[] args) throws Exception {
             throw new IllegalStateException("No namespaces were provided");
         }
 
-        List<Namespace> n = new ArrayList<>();
-        for(String s : namespaces){
-            if("all".equalsIgnoreCase(s)){
-                Collections.addAll(n, Namespace.values());
-                break;
+        try {
+            if (projects.contains("nd4j")) {
+                generateNamespaces(NS_PROJECT.ND4J, outputDir, "org.nd4j.linalg.factory");
             }
-
-            Namespace ns = Namespace.fromString(s);
-            if(ns == null){
-                throw new IllegalStateException("Invalid/unknown namespace provided: " + s);
+            if (projects.contains("sd")) {
+                generateNamespaces(NS_PROJECT.SAMEDIFF, outputDir, "org.nd4j.autodiff.samediff");
             }
```

Comment:
Minor UX thing here: we should probably do 1 of 2 things:
(a) Make `projects` arg required on the command line, OR
(b) Have a default value (probably both ND4J and SameDiff)

I'm just thinking that at present, if no `-projects ...` arg is passed to the CLI, we'll get nothing generated, and no warning/error etc.

2 other UX things:
(a) it's case sensitive
(b) no validation for something like `-projects thisDoesntExist`

Reply:
I'd also add "-projects all" to make options consistent. And "all" can be a default value for both -projects and -namespaces - to avoid running command when no output can be produced.

Reply:
Default value "all" and validator added.